### ISSUE TITLE
Package files in the future handling 

### ISSUE THREAD

**Issue by yishaigalatzer** [NuGet/Home#2794]
Some packages have files with no dates, or dates in the future.

The current behavior in NuGet 3.4.3 is that these packages are extracted with dates as is (in the future), and that makes build rebuild every time, because it compares timestamps.

We do not want to change all the dates to the current datetime, because it will break deployment scenarios and might have other undesirable side effects. So what we decided to do is extract dates in the future with datetime.now (similar to how we extract files without dates).

In a nutshell use: `file.LastWriteTime > DateTime.Now then use DateTime.Now`

We also want to warn during the `pack` operation if a file in future exists (this action will wait for the next release)

**Comment by rrelyea** [comment]
@spadapet -Confirmed that we are targeting 3.4.4 with this fix.

**Comment by emgarten** [comment]
Verified in 3.4.4
